src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -884,6 +884,19 @@ public virtual int GetByteCount(String s)
         [Pure]
         public abstract int GetByteCount(char[] chars, int index, int count);
 
+        // Returns the number of bytes required to encode a string range.
+        //
+        [Pure]
+        public virtual int GetByteCount(string s, int index, int count)
+        {
+            if (s == null)
```

Comment:
although I prefer validating the parameters in this method so the thrown exception can be more clear about the problem but I see the perf benefit too. so I am ok to keep what you have but please add assert in the methods you are introducing 

---
In reply to: [92667264](https://github.com/dotnet/coreclr/pull/8651#discussion_r92667264) [](ancestors = 92667264)